app/views/admin/AdministerUserModal.coffee

```diff
@@ -193,6 +201,53 @@ module.exports = class AdministerUserModal extends ModelModal
       @state = 'made-prepaid'
       @renderSelectors('#prepaid-form')
 
+  onClickAddEsportsProductButton: ->
+    attrs = forms.formToObject(@$('#esports-product-form'))
+
+    return unless _.all(_.values(attrs))
+    return unless attrs.endDate and attrs.startDate and attrs.endDate > attrs.startDate
+    attrs.endDate = attrs.endDate + " " + "23:59"   # Otherwise, it ends at 12 am by default which does not include the date indicated
+
+    attrs.startDate = moment.timezone.tz(attrs.startDate, @timeZone ).toISOString()
+    attrs.endDate = moment.timezone.tz(attrs.endDate, @timeZone).toISOString()
+
+    attrs.productOptions = {type: attrs.esportsType, id: _.uniqueId()}
```

Comment:
what is the use of id here?

Reply:
we're going to have edit feature in administrator-user modal to edit the esports product. so add an id here to make sure we can edit the correct product. (maybe we can use array index too but since we have mix types of products in user.products so i think index may not so correct to use)